packages/rollup/src/index.js

```diff
@@ -0,0 +1,258 @@
+  }
+
+  // Does NOT support (unlike rollup CLI):
+  // * factory function
+  // * multiple configs for multiple outputs
+  if (Array.isArray(config) || typeof config === 'function') {
+    throw new Error('Arrays + factory configs unsupported');
+  }
+
+  return config;
+}
+
+// Processing of --environment CLI options into environment vars
+// https://github.com/rollup/rollup/blob/v1.31.0/cli/run/index.ts#L50-L57
+function extractEnvironmentVariables(vars) {
+  vars.split(',').forEach(pair => {
+    const [key, ...value] = pair.split(':');
+    if (value.length) {
+      process.env[key] = value.join(':');
+    } else {
+      process.env[key] = String(true);
+    }
+  });
+}
+
+// Parse a subset of supported CLI arguments required for the rollup_bundle rule API.
+// Returns input/outputOptions for the rollup.bundle/write() API
+//  input:  https://rollupjs.org/guide/en/#inputoptions-object
+//  output: https://rollupjs.org/guide/en/#outputoptions-object
+async function parseCLIArgs(args) {
+  let inputOptions = {
+    onwarn(...warnArgs) {
+      worker.log(...warnArgs);
+    },
+  };
+
+  let outputOptions = {};
+
+  let configFile = null;
+
+  // Input files to rollup
+  let inputs = [];
+
+  // Followed by suppported rollup CLI options
+  for (let i = 0; i < args.length; i++) {
+    const arg = args[i];
+
+    // Non-option is assumed to be an input file
+    if (!arg.startsWith('--')) {
+      inputs.push(arg);
+      continue;
+    }
+
+    const option = arg.slice(2);
+    switch (option) {
+      case 'config':
+        configFile = path.resolve(args[++i]);
+        break;
+
+      case 'silent':
+        inputOptions.onwarn = () => {};
+        break;
+
+      case 'output.dir':
+      case 'output.file':
+      case 'format':
+      case 'sourcemap':
+        outputOptions[option.replace('output.', '')] = args[++i];
+        break;
+
+      case 'preserveSymlinks':
+        inputOptions[option] = true;
+        break;
+
+      // Common rollup CLI args, but not required for use
+      case 'environment':
+        extractEnvironmentVariables(args[++i]);
+        break;
+
+      default:
+        throw new Error(`${MNEMONIC}: invalid or unsupported argument ${arg}`);
+    }
+  }
+
+  // If outputting a directory then rollup_bundle.bzl passed a series
+  // of name=path files as the input.
+  // TODO: do some not have the =?
+  if (outputOptions.dir) {
+    inputs = inputs.reduce((m, nameInput) => {
+      const [name, input] = nameInput.split('=', 2);
+      m[name] = input;
+      return m;
+    }, {});
+  }
+
+  // Additional options passed via config file
+  if (configFile) {
+    const config = await loadConfigFile(configFile);
+
+    if (config.output) {
+      outputOptions = {...config.output, ...outputOptions};
+    }
+
+    inputOptions = {...config, ...inputOptions};
+
+    // Delete from our copied inputOptions, not the config which
+    // may be external and persisted across runs
+    delete inputOptions.output;
```

Comment:
@alexeagle this was the main problem with config files being lost... previously we deleted `config.output` in the  `if (config.output)`. That `config` object gets cached, so next time we run the test the worker has bad cache data.

Now we simply delete it from our per-run instance of the options.

Reply:
cool nice find